Odd behavior for .and_assert_called_ordered()

**I'm using `TestSlide` version:** 2.6.1

**Given:**
```python
# repro.py
import testslide
import os


class Repro(testslide.TestCase):
    def test_repro(self):
        self.mock_callable(os, "kill").for_call(0, 0).to_return_value(
            1
        ).and_assert_called_ordered()
        self.mock_callable(os, "kill").for_call(0, 0).to_return_value(
            2
        ).and_assert_called_ordered()

        print(os.kill(0, 0))
        print(os.kill(0, 0))
```

**When I run:**
```
 testslide repro.py 
```

**I expected this to happen:**

For the test to pass.

**But, instead this happened:**

it fails:

```
$ testslide repro.py 
repro.Repro
2
2
  test_repro: AssertionError: calls did not match assertion.

Failures:

  1) repro.Repro: test_repro
    1) AssertionError: calls did not match assertion.
    
    These calls were expected to have happened in order:
    
      <module 'os' from '/home/fornellas/.pyenv/versions/3.8.6/lib/python3.8/os.py'>, 'kill' with arguments:
        (0, 0)
      <module 'os' from '/home/fornellas/.pyenv/versions/3.8.6/lib/python3.8/os.py'>, 'kill' with arguments:
        (0, 0)
    
    but these calls were made:
    
      <module 'os' from '/home/fornellas/.pyenv/versions/3.8.6/lib/python3.8/os.py'>, 'kill' with arguments:
        (0, 0)
      <module 'os' from '/home/fornellas/.pyenv/versions/3.8.6/lib/python3.8/os.py'>, 'kill' with arguments:
        (0, 0)

Finished 1 example(s) in 0.6s 
  Failed: 1
```
It seems there's a confusion with semantics here:

- `.for_call()` is stacked, and the last one that says "I can answer to this call" runs.
  - In this case, as both `.for_call` match the arguments, only the last one is executed for the 2 calls.
- `.and_assert_called_ordered()` gives the impression that this would work, as 2 calls happened.
  - It does not, as both `.for_call()` accepts the same arguments.
  - Only the last one is executed.
  - The error message does not make it clear that the 2 calls made were for the last `.for_call()`.

--- a/tests/mock_callable_testslide.py
+++ b/tests/mock_callable_testslide.py
@@ -982,6 +982,17 @@ def mock_callable_tests(context):
     ##
 
     @context.sub_context
+    def stacked_same_for_call(context):
+        @context.example
+        def it_works_at_declared_order(self):
+            self.mock_callable(os, "getpgid").for_call(1).to_return_value(1)
+            self.mock_callable(os, "getpgid").for_call(1).to_return_value(2)
+
+            self.assertEqual(os.getpgid(1), 1)
+            self.assertEqual(os.getpgid(1), 2)
+            self.assertEqual(os.getpgid(1), 2)
+
+    @context.sub_context
     def call_order_assertion(context):
         @context.memoize
         def target1(self):
@@ -1077,6 +1088,22 @@ def mock_callable_tests(context):
             self.assertEqual(self.target1.f2("step 2"), "step 2 return")
             self.assertEqual(self.target2.f1("step 3"), "step 3 return")
             self.assert_all()
+
+        @context.example
+        def stacked_same_for_call_works(self):
+            self.mock_callable(os, "getpgid").for_call(1).to_return_value(
+                1
+            ).and_assert_called_ordered()
+            self.mock_callable(os, "getpgid").for_call(2).to_return_value(
+                2
+            ).and_assert_called_ordered()
+            self.mock_callable(os, "getpgid").for_call(1).to_return_value(
+                3
+            ).and_assert_called_ordered()
+
+            self.assertEqual(os.getpgid(1), 1)
+            self.assertEqual(os.getpgid(2), 2)
+            self.assertEqual(os.getpgid(1), 3)
 
     @context.sub_context
     def when_target_is_a_module(context):

--- a/testslide/mock_callable.py
+++ b/testslide/mock_callable.py
@@ -551,10 +551,25 @@ class _CallableMock(object):
         self.is_async = is_async
         self.type_validation = type_validation
         self.caller_frame_info = caller_frame_info
+        self.called_runner_indexes: List[int] = []
 
     def _get_runner(self, *args: Any, **kwargs: Any) -> Any:
-        for runner in self.runners:
+        for index, runner in enumerate(self.runners):
             if runner.can_accept_args(*args, **kwargs):
+                if runner.accepted_args:
+                    same_call_runners = [
+                        other_runner
+                        for other_runner in self.runners[index + 1 :]
+                        if other_runner.accepted_args == runner.accepted_args
+                    ]
+                    if same_call_runners:
+                        for final_runner in reversed([runner] + same_call_runners):
+                            if (
+                                self.runners.index(final_runner)
+                                not in self.called_runner_indexes
+                            ):
+                                return final_runner
+                        return runner
                 return runner
         return None
 
@@ -571,6 +586,9 @@ class _CallableMock(object):
     def __call__(self, *args: Any, **kwargs: Any) -> Optional[Any]:
         runner = self._get_runner(*args, **kwargs)
         if runner:
+            index = self.runners.index(runner)
+            if index not in self.called_runner_indexes:
+                self.called_runner_indexes.append(index)
             if self.is_async:
                 if isinstance(runner, _AsyncRunner):
 

